packages/plugin-metrics/src/metrics.js

```diff
@@ -26,16 +26,75 @@ const Metrics = SparkPlugin.extend({
     return this.batcher.request(Object.assign({key}, value));
   },
 
-  submitClientMetrics(eventName, props) {
+  /**
+   * This corresponds to #sendSemiStructured() in the deprecated metrics handler
+   * @param eventName
+   * @param props
+   * @param preLoginId
+   */
+  submitClientMetrics(eventName, props, preLoginId) {
```

Comment:
This should be singular. It's also slightly redundant. Any way we can avoid saying metrics twice in `spark.metrics.submitClientMetrics`?